milli/src/index.rs

```diff
@@ -1185,43 +1188,88 @@ pub(crate) mod tests {
     }
 
     impl TempIndex {
-        /// Creates a temporary index, with a default `4096 * 100` size. This should be enough for
-        /// most tests.
-        pub fn new() -> Self {
+        /// Creates a temporary index
+        pub fn new_with_map_size(size: usize) -> Self {
             let mut options = EnvOpenOptions::new();
-            options.map_size(100 * 4096);
+            options.map_size(size);
             let _tempdir = TempDir::new_in(".").unwrap();
             let inner = Index::new(options, _tempdir.path()).unwrap();
-            Self { inner, _tempdir }
+            let indexer_config = IndexerConfig::default();
+            let index_documents_config = IndexDocumentsConfig::default();
+            Self { inner, indexer_config, index_documents_config, _tempdir }
+        }
+        /// Creates a temporary index, with a default `4096 * 1000` size. This should be enough for
+        /// most tests.
+        pub fn new() -> Self {
+            Self::new_with_map_size(4096 * 1000)
+        }
+        pub fn add_documents_using_wtxn<'t, R>(
+            &'t self,
+            wtxn: &mut RwTxn<'t, '_>,
+            documents: DocumentsBatchReader<R>,
+        ) -> Result<(), crate::error::Error>
+        where
+            R: std::io::Read + std::io::Seek,
+        {
+            let builder = IndexDocuments::new(
+                wtxn,
+                &self,
+                &self.indexer_config,
+                self.index_documents_config.clone(),
+                |_| (),
+            )
+            .unwrap();
+            let (builder, user_error) = builder.add_documents(documents).unwrap();
+            user_error?;
+            builder.execute()?;
+            Ok(())
+        }
+        pub fn add_documents<R>(
+            &self,
+            documents: DocumentsBatchReader<R>,
+        ) -> Result<(), crate::error::Error>
+        where
+            R: std::io::Read + std::io::Seek,
+        {
+            let mut wtxn = self.write_txn().unwrap();
+            self.add_documents_using_wtxn(&mut wtxn, documents)?;
+            wtxn.commit().unwrap();
+            Ok(())
+        }
+
+        pub fn update_settings(
+            &self,
+            update: impl Fn(&mut Settings),
+        ) -> Result<(), crate::error::Error> {
+            let mut wtxn = self.write_txn().unwrap();
+            self.update_settings_using_wtxn(&mut wtxn, update)?;
+            wtxn.commit().unwrap();
+            Ok(())
+        }
+        pub fn update_settings_using_wtxn<'t>(
```

Comment:
```suggestion
        }
        
        pub fn update_settings_using_wtxn<'t>(
```
did you configure rustfmt on your editor?

Reply:
I did, but for some reason every once in a while it doesn't work for a file :/